### PR TITLE
Improve scale up memory allocation

### PR DESCRIPTION
**Pull request by obino**
This PR takes in account the potential uses of memory for AppServers considering the current setting of max_memory. It prevents overcommit memory.

### CODE REVIEW

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2395/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2396/

File: InfrastructureManager/system_manager.py [hunk 1]
```diff
@@ -102,7 +103,8 @@ def get_memory_usage(self, secret):
     mem_stats_dict = { JSONTags.MEMORY :
       {
         JSONTags.AVAILABLE : mem_stats.available,
-        JSONTags.USED : mem_stats.used
+        JSONTags.USED : mem_stats.used,
+        JSONTags.TOTAL : mem_stats.total
```
Comment (menivaitsi) [inline InfrastructureManager/system_manager.py]: Please remove the changes in the SystemManager. They should be part of the shift in using `get_all_stats` in AppController.

File: InfrastructureManager/tests/test_system_manager.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def test_get_memory_usage(self):
     flexmock(utils).should_receive("get_secret").and_return("fake secret")
 
     expected_keys = [
-      JSONTags.MEMORY, JSONTags.AVAILABLE, JSONTags.USED
+      JSONTags.MEMORY, JSONTags.AVAILABLE, JSONTags.USED, JSONTags.TOTAL
```
Comment (menivaitsi) [inline InfrastructureManager/tests/test_system_manager.py]: Same here.

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2398/

**Comment by obino** [comment]
OCD run https://ocd.appscale.com:8080/job/Daily%20Build/2399/

**Comment by obino** [comment]
OCD job for restart https://ocd.appscale.com:8080/job/Euca-Faststart/473/console

**Pull request merged**